roles/variables/defaults/main.yml

```diff
@@ -1452,6 +1466,39 @@ sso_groups_scope: none
 ### Configures whether offline_access scope would be requested in the authorization URI, Set this to false if offline tokens are not allowed for the user or client in IdP
 sso_refresh_token: true
 
+### Boolean to enable OAuth on all cp components, except ksqlDB. When set to true, you must set oauth_jwks_uri, oauth_authorize_uri, oauth_token_uri, oauth_issuer_url, oauth_client_id, oauth_client_password
+oauth_enabled: false
+
+### Boolean to enable OAuth with LDAP, required when dual support needed. oauth_enabled flag also needs to be true
+ldap_with_oauth_enabled: false
+
+### Client id for authorize and token request to Idp via cp components. This is the super user for all MDS api calls
+oauth_client_id: none
+
+### Client password for authorize and token request to Idp via cp components
+oauth_client_password: none
+
+### IdP token endpoint, from where a token is requested by MDS when OAuth is enabled
+oauth_token_uri: none
+
+### The issuer url, which is typically the authorization server's URL. This value is used to compare to issuer claim in the JWT token for verification
+oauth_issuer_url: none
+
+### The OAuth/OIDC provider URL from which the provider's JWKS (JSON Web Key Set) can be retrieved.
+oauth_jwks_uri: none
```

Comment:
Do we have any check to verify the mandatory variables. Lets say 
if oauth_enabled is true oath_jwks_url cannot be null

Reply:
we don't have such checks as of now, the plan is to add this in the documentation. Should we add such a validation task here? Historically, we don't validate the input in inventory file